File: 01_gpio_digital/led_blink_task.c
#include <wiringPi.h>
#define LED_PIN 7 
#define YELLOW_PIN 5
#define GREEN_PIN 4

int main (void)
{
    wiringPiSetup () ;
    pinMode (LED_PIN, OUTPUT) ;
    pinMode (YELLOW_PIN, OUTPUT) ;
    pinMode (GREEN_PIN, OUTPUT) ;
    digitalWrite (LED_PIN, HIGH) ;  
    delay (2000) ;
    digitalWrite (LED_PIN, LOW) ;  
    digitalWrite (YELLOW_PIN, HIGH) ;  
    delay (2000) ;
    digitalWrite (YELLOW_PIN, LOW) ;
    digitalWrite (GREEN_PIN, HIGH) ;  
    delay (2000) ;
    digitalWrite (GREEN_PIN, LOW) ;  
    delay (2000) ;
   return 0;
}

File: 01_gpio_digital/led_blink_task.py
import RPi.GPIO as GPIO
import time

RED_PIN = 4
YELLOW_PIN = 24
GREEN_PIN = 23

GPIO.setmode(GPIO.BCM)
GPIO.setup(RED_PIN, GPIO.OUT)
GPIO.setup(YELLOW_PIN, GPIO.OUT)
GPIO.setup(GREEN_PIN, GPIO.OUT)


GPIO.output(RED_PIN, GPIO.HIGH)
print("led on")
time.sleep(2)
GPIO.output(RED_PIN, GPIO.LOW)
print("led off")
GPIO.output(YELLOW_PIN, GPIO.HIGH)
print("led on")
time.sleep(2)
GPIO.output(YELLOW_PIN, GPIO.LOW)
print("led off")
GPIO.output(GREEN_PIN, GPIO.HIGH)
print("led on")
time.sleep(2)
GPIO.output(GREEN_PIN, GPIO.LOW)
print("led off")


GPIO.cleanup()